Core renaming

VladimirIvan opened the pull request:
- Updaed license text.
- Fixed header guards (`PACKAGE_NAME_CLASS_NAME_H`).
- Put `KinematicElement` and `Printable` into `exotica` namespace.
- Changed comments to `//` and doxygen comments to `///`.
- Updated function and private member names names.
- Changed initializer naming:
    - `file_name.in` and generated `file_name.h` have will match the `AddInitializer(file_name)`.
    - The class name no longer has to match the file name. Specify the class name using `class ClassName` inside the `*.in` file. The class name has to match the class that is being initialized (same as before).
    - `include` and `extend` now have to use the file name: `extend <package_name/file_name>`.
    - All member variables of the initializer (in c++) are now lower_case_with_underscores. The names are converted from CamelCased names at compile time. XML and Python dictionaries use the camel cased names.
 - A few member names use incorrect convention but only when they derive from a parent class outside of EXOTica.
 - Public members like `J`, `JN`, `startJ`, `H` have been renamed to `jacobian`, `length_jacobian`, `start_jacobian`, `hessian`. Private math based symbols were left as they were (`W`, `b`, `B`, ...).
 - Tested, checked all examples, tested with SNOPT.

This is a big one. I tried to split the initializers from the rest but they affect 80% of the files so you'd have to review the same code twice. I hope this will be faster.

wxmerkt commented on exotations/exotica_core_task_maps/src/task_map_py.cpp:
```diff
@@ -68,15 +96,15 @@ PYBIND11_MODULE(exotica_core_task_maps_py, module)
-        .def_property("W", &IMesh::get_weights, &IMesh::set_weights)
-        .def("set_weight", &IMesh::set_weight)
+        .def_property("W", &IMesh::GetWeights, &IMesh::SetWeights)
+        .def("SetWeight", &IMesh::SetWeight)
```
Nope, this is incorrect please revert. Python is methods with underscore (I had this fixed in the earlier one)

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/exotica_core_task_maps/test/test_maps.cpp:
```diff
@@ -1110,7 +1107,7 @@ TEST(ExoticaTaskMaps, testJointSmoothingBackwardDifference)
 TEST(ExoticaTaskMaps, testLookAt)
 {
-    try
+    // try
```
Why is this commented?

wxmerkt replied:
Unresolved - why is this commented? Will it now fail otherwise?

wxmerkt commented on exotations/exotica_core_task_maps/test/test_maps.cpp:
```diff
@@ -1127,15 +1124,15 @@ TEST(ExoticaTaskMaps, testLookAt)
-    {
-        ADD_FAILURE() << "Uncaught exception!";
-    }
+    // catch (...)
```
Why is this commented?

wxmerkt replied:
As above, why commented?

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -1,66 +1,53 @@
-#define EXOTICA_AICO_SOLVER_AICO_SOLVER_H_
+// Copyright (c) 2018, University of Edinburgh
+// All rights reserved.
+//
```
You removed the note on which reference / original code / publication it is based

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -70,61 +57,52 @@
+    UnconstrainedTimeIndexedProblemPtr prob_;  //!< Shared pointer to the planning problem.
+    double damping;                            //!< Damping
+    double damping_init_;                      //!< Damping
+    double minimum_step_tolerance;             //!< Update tolerance to stop update of messages if change of maximum coefficient is less than this tolerance.
```
Why did you remove the `_` for those three tolerance variables?

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -70,61 +57,52 @@
+    double damping;                            //!< Damping
+    double damping_init_;                      //!< Damping
+    double minimum_step_tolerance;             //!< Update tolerance to stop update of messages if change of maximum coefficient is less than this tolerance.
+    double step_tolerance;                     //!< Relative step tolerance (termination criterion)
```
As above

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -70,61 +57,52 @@
+    double damping_init_;                      //!< Damping
+    double minimum_step_tolerance;             //!< Update tolerance to stop update of messages if change of maximum coefficient is less than this tolerance.
+    double step_tolerance;                     //!< Relative step tolerance (termination criterion)
+    double function_tolerance;                 //!< Relative function tolerance/first-order optimality criterion
```
As above

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -181,103 +159,86 @@ class AICOSolver : public MotionSolver, public Instantiable<AICOSolverInitialize
+}
 
-#endif /* EXOTICA_AICO_SOLVER_AICO_SOLVER_H_ */
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/aico_solver.h:
```diff
@@ -181,103 +159,86 @@ class AICOSolver : public MotionSolver, public Instantiable<AICOSolverInitialize
     double Step();
 };
-} /* namespace exotica */
+}
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/bayesian_ik_solver.h:
```diff
@@ -48,61 +37,52 @@
+    UnconstrainedEndPoseProblemPtr prob_;  //!< Shared pointer to the planning problem.
+    double damping;                        //!< Damping
+    double damping_init_;                  //!< Damping
+    double minimum_step_tolerance;         //!< Update tolerance to stop update of messages if change of maximum coefficient is less than this tolerance.
```
As above, why are these three member variables changed?

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/bayesian_ik_solver.h:
```diff
@@ -151,98 +131,79 @@ class BayesianIKSolver : public MotionSolver, public Instantiable<BayesianIKSolv
     double Step();
 };
-} /* namespace exotica */
+}
```
Please revert and keep as in original

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/bayesian_ik_solver.h:
```diff
@@ -151,98 +131,79 @@ class BayesianIKSolver : public MotionSolver, public Instantiable<BayesianIKSolv
+}
 
-#endif /* EXOTICA_AICO_SOLVER_BAYESIAN_IK_SOLVER_H_ */
+#endif
```
Please revert and keep as in original

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/incremental_gaussian.h:
```diff
@@ -168,4 +161,4 @@ class SinglePassMeanCovariance
 };
 
-#endif /* EXOTICA_AICO_SOLVER_INCREMENTAL_GAUSSIAN_H_ */
+#endif
```
Please revert and keep as in original

wxmerkt commented on exotations/solvers/exotica_aico_solver/include/exotica_aico_solver/math_operations.h:
```diff
@@ -1,69 +1,54 @@
 }
 
-#endif  // EXOTICA_AICO_SOLVER_MATH_OPERATIONS_H_
+#endif
```
Please revert and keep as in original

wxmerkt commented on exotations/solvers/exotica_aico_solver/src/aico_solver.cpp:
```diff
@@ -1,43 +1,38 @@
+// POSSIBILITY OF SUCH DAMAGE.
+//
+
+// This code is based on algorithm developed by Marc Toussaint
```
These notes are notes are missing in the header, please copy there.

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/src/aico_solver.cpp:
```diff
@@ -647,4 +642,4 @@ void AICOSolver::PerhapsUndoStep()
     }
 }
-} /* namespace exotica */
+}
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/src/bayesian_ik_solver.cpp:
```diff
@@ -339,18 +334,18 @@ double BayesianIKSolver::GetTaskCosts()
+    for (int i = 0; i < prob_->cost.num_tasks; i++)
     {
-        prec = prob_->Cost.Rho(i);
+        prec = prob_->cost.rho(i);
```
Rho changes to rho - i.e., change from upper to lower case; this is technically incorrect if it does not match our book chapter or paper.

wxmerkt commented on exotations/solvers/exotica_aico_solver/src/bayesian_ik_solver.cpp:
```diff
@@ -520,4 +515,4 @@ void BayesianIKSolver::PerhapsUndoStep()
     }
 }
-} /* namespace exotica */
+}
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_ik_solver/include/exotica_ik_solver/ik_solver.h:
```diff
@@ -61,18 +59,18 @@ class IKSolver : public MotionSolver, public Instantiable<IKSolverInitializer>
 }
 
-#endif /* EXOTICA_IK_SOLVER_IK_SOLVER_H_ */
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_levenberg_marquardt_solver/include/exotica_levenberg_marquardt_solver/levenberg_marquardt_solver.h:
```diff
@@ -14,15 +43,15 @@ class LevenbergMarquardtSolver : public MotionSolver, public Instantiable<Levenb
+}
 
-#endif  // EXOTICA_LEVENBERG_MARQUARDT_SOLVER_LEVENBERG_MARQUARDT_SOLVER_H_
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_levenberg_marquardt_solver/include/exotica_levenberg_marquardt_solver/levenberg_marquardt_solver.h:
```diff
@@ -14,15 +43,15 @@ class LevenbergMarquardtSolver : public MotionSolver, public Instantiable<Levenb
     double lambda_ = 0;  ///< Damping factor
 };
-}  // namespace exotica
+}
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_levenberg_marquardt_solver/src/levenberg_marquardt_solver.cpp:
```diff
@@ -78,44 +106,43 @@ void LevenbergMarquardtSolver::Solve(Eigen::MatrixXd& solution)
-    planning_time_ = timer.getDuration();
+    planning_time_ = timer.GetDuration();
+}
 }
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_ompl_solver/CMakeLists.txt:
```diff
@@ -14,7 +14,7 @@ catkin_package(
 )
 
-AddInitializer(OMPLSolver RRT RRTConnect PRM LazyPRM EST KPIECE BKPIECE RRTStar LBTRRT)
+AddInitializer(ompl_solver rrt rrt_connect prm lazy_prm est kpiece bkpiece rrt_star lbt_rrt)
```
When changing, please make onto new lines. Also, this makes established abbreviations lower case.

wxmerkt commented on exotations/solvers/exotica_ompl_solver/include/exotica_ompl_solver/ompl_exo.h:
```diff
@@ -297,7 +292,6 @@ class OMPLSE2RNProjection : public ompl::base::ProjectionEvaluator
-} /* namespace exotica */
-
-#endif /* EXOTICA_OMPL_SOLVER_OMPL_EXO_H_ */
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_ompl_solver/include/exotica_ompl_solver/ompl_exo.h:
```diff
@@ -297,7 +292,6 @@ class OMPLSE2RNProjection : public ompl::base::ProjectionEvaluator
 private:
     std::vector<int> variables_;
 };
+}
```
Please revert

wxmerkt replied:
Unresolved

wxmerkt commented on exotations/solvers/exotica_ompl_solver/include/exotica_ompl_solver/ompl_native_solvers.h:
```diff
@@ -105,19 +100,19 @@ class LazyPRM : public OMPLSolver<SamplingProblem>, Instantiable<LazyPRMInitiali
 }
 
-#endif /* EXOTICA_OMPL_SOLVER_OMPL_NATIVE_SOLVER_H_ */
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_ompl_solver/include/exotica_ompl_solver/ompl_solver.h:
```diff
@@ -102,4 +97,4 @@ class OMPLSolver : public MotionSolver
 }
 
-#endif /* EXOTICA_OMPL_SOLVER_OMPL_SOLVER_H_ */
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_ompl_solver/src/ompl_solver.cpp:
```diff
@@ -210,30 +206,30 @@ void OMPLSolver<ProblemType>::Solve(Eigen::MatrixXd &solution)
     }
 
-    SetGoalState(prob_->getGoalState(), init_.Epsilon);
+    SetGoalState(prob_->GetGoalState(), init_.epsilon);
```
Epsilon also changes from upper to lower case. In this case, though, it is an improvement (as for alpha in the IMesh taskmap) as it fixes a previously incorrect case.

wxmerkt commented on exotations/solvers/exotica_time_indexed_rrt_connect_solver/include/exotica_time_indexed_rrt_connect_solver/time_indexed_rrt_connect.h:
```diff
@@ -220,18 +217,18 @@ class OMPLTimeIndexedRRTConnect : public base::Planner
     double forwardTimeDistance(const Motion *a, const Motion *b) const
     {
-        static const Eigen::VectorXd max_vel = si_->getStateSpace()->as<OMPLTimeIndexedRNStateSpace>()->prob_->vel_limits_;
+        static const Eigen::VectorXd max_vel = si_->getStateSpace()->as<OMPLTimeIndexedRNStateSpace>()->prob_->vel_limits;
```
Why did this private member variable change to without `_`?

wxmerkt commented on exotations/solvers/exotica_time_indexed_rrt_connect_solver/include/exotica_time_indexed_rrt_connect_solver/time_indexed_rrt_connect.h:
```diff
@@ -284,33 +281,34 @@ class OMPLTimeIndexedRRTConnect : public base::Planner
 }
-#endif /* TIME_INDEXED_RRT_CONNECT_SOLVER_TIME_INDEXED_RRT_CONNECT_H_ */
+
+#endif
```
Please revert

wxmerkt commented on exotations/solvers/exotica_time_indexed_rrt_connect_solver/src/time_indexed_rrt_connect.cpp:
```diff
@@ -1,55 +1,53 @@
     getSubspace(0)->as<ompl::base::RealVectorStateSpace>()->setBounds(bounds);
     addSubspace(ompl::base::StateSpacePtr(new ompl::base::TimeStateSpace), 1.0);
-    getSubspace(1)->as<ompl::base::TimeStateSpace>()->setBounds(0, prob_->T);
+    getSubspace(1)->as<ompl::base::TimeStateSpace>()->setBounds(0, prob_->GetGoalTime());
```
Is this a bug fix? `T` is the number of timesteps (or not?!) while goal time is something else. I think this is a hidden fix.

wxmerkt replied:
Unresolved

wxmerkt replied:
This is something we should be having a look at. It is definitely different now than before - GetGoalTime != GetT

wxmerkt commented on exotica_core/cmake/generate_initializers.py:
```diff
@@ -232,152 +280,196 @@ def ParseLine(line, ln, fn):
-            ClassName = os.path.basename(sys.argv[offset+i][0:-3])
-            with open(Input,"r") as fi:
-                with open(DevelDir+'/init/'+ClassName+'.in',"w") as f:
+    if len(sys.argv) > 5:
```
What is the extra argument here?

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/collision_scene.h:
```diff
@@ -271,26 +263,25 @@ class CollisionScene : public Uncopyable
 }
 
-#endif  // COLLISIONSCENE_H
+#endif
```
Please revert

wxmerkt commented on exotica_core/include/exotica_core/exotica_core.h:
```diff
@@ -49,4 +44,4 @@
 #include <exotica_core/version.h>
 
-#endif  // EXOTICA_H
+#endif
```
Ending header guard comment please

wxmerkt commented on exotica_core/include/exotica_core/kinematic_element.h:
```diff
@@ -1,99 +1,130 @@
 {
 public:
-    KinematicElement(int id, std::shared_ptr<KinematicElement> parent, const KDL::Segment& segment) : Parent(parent), Segment(segment), Id(id)
+    KinematicElement(int _id, std::shared_ptr<KinematicElement> _parent, const KDL::Segment& _segment) : parent(_parent), segment(_segment), id(_id)
```
Shouldn't this be `id_in` or is this from the style guide?

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/kinematic_element.h:
```diff
@@ -1,99 +1,130 @@
+}
 
-#endif  // KINEMATICELEMENT_H
+#endif
```
Ending header guard comment

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
-    std::string FrameBLinkName;
-    KDL::Frame FrameBOffset;
+    KinematicFrameRequest(std::string _frame_a_link_name, KDL::Frame _frame_a_offset = KDL::Frame(), std::string _frame_b_link_name = "", KDL::Frame _frame_b_offset = KDL::Frame());
+    std::string frame_a_link_name;
```
`frame_A_link_name`, `frame_B_link_name` etc. - A and B are identifiers as used in papers etc. and should be kept capitalised as is (applies to everywhere)

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
+    KinematicResponse(KinematicRequestFlags _flags, int _size, int _N = 0);
+    KinematicRequestFlags flags = KinematicRequestFlags::KIN_FK;
+    std::vector<KinematicFrame> frame;
+    Eigen::VectorXd x;
```
X should remain upper case. It is the vector (thus upper case) of configuration values (as in papers)

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
+    KinematicRequestFlags flags = KinematicRequestFlags::KIN_FK;
+    std::vector<KinematicFrame> frame;
+    Eigen::VectorXd x;
+    ArrayFrame phi;
```
Phi should remain upper case. It is the forward kinematic mapping as in papers.

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
-    Eigen::Map<ArrayTwist> PhiDot{nullptr, 0};
-    Eigen::Map<ArrayJacobian> J{nullptr, 0};
-    Eigen::Map<ArrayJacobian> JDot{nullptr, 0};
+    Eigen::Map<ArrayFrame> phi{nullptr, 0};
```
Phi should remain upper case.

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
-    KDL::Frame FK(const std::string& elementA, const KDL::Frame& offsetA, const std::string& elementB, const KDL::Frame& offsetB);
-    Eigen::MatrixXd Jacobian(std::shared_ptr<KinematicElement> elementA, const KDL::Frame& offsetA, std::shared_ptr<KinematicElement> elementB, const KDL::Frame& offsetB);
-    Eigen::MatrixXd Jacobian(const std::string& elementA, const KDL::Frame& offsetA, const std::string& elementB, const KDL::Frame& offsetB);
+    KDL::Frame FK(std::shared_ptr<KinematicElement> element_a, const KDL::Frame& offset_a, std::shared_ptr<KinematicElement> element_b, const KDL::Frame& offset_b);
```
A and B should remain upper case

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_end_pose_problem.h:
```diff
@@ -52,31 +46,31 @@ class BoundedEndPoseProblem : public PlanningProblem, public Instantiable<Bounde
+    double GetScalarTaskCost(const std::string& task_name);
 
-    EndPoseTask Cost;
+    EndPoseTask cost;
```
Should this keep an `_`?

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_end_pose_problem.h:
```diff
@@ -52,31 +46,31 @@ class BoundedEndPoseProblem : public PlanningProblem, public Instantiable<Bounde
-    TaskSpaceVector Phi;
-    Hessian H;
-    Eigen::MatrixXd J;
+    TaskSpaceVector phi;
```
Should retain upper case?

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_time_indexed_problem.h:
```diff
@@ -1,105 +1,99 @@
-    TimeIndexedTask Cost;
-    TaskSpaceVector CostPhi;
+    TimeIndexedTask cost;
+    TaskSpaceVector cost_phi;
```
Upper case Phi

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_time_indexed_problem.h:
```diff
@@ -1,105 +1,99 @@
+    TaskSpaceVector cost_phi;
 
-    double W_rate;  //!< Kinematic system transition error covariance multiplier (constant throughout the trajectory)
+    double w_rate_;  //!< Kinematic system transition error covariance multiplier (constant throughout the trajectory)
```
W is a math quantity

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_time_indexed_problem.h:
```diff
@@ -1,105 +1,99 @@
-    std::vector<TaskSpaceVector> Phi;
-    std::vector<Eigen::MatrixXd> J;
-    std::vector<Hessian> H;
+    std::vector<TaskSpaceVector> phi;
```
Phi upper-case

wxmerkt commented on exotica_core/include/exotica_core/problems/end_pose_problem.h:
```diff
@@ -50,48 +45,48 @@ class EndPoseProblem : public PlanningProblem, public Instantiable<EndPoseProble
-    TaskSpaceVector Phi;
-    Eigen::MatrixXd J;
-    Hessian H;
+    TaskSpaceVector phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/end_pose_problem.h:
```diff
@@ -50,48 +45,48 @@ class EndPoseProblem : public PlanningProblem, public Instantiable<EndPoseProble
+    bool use_bounds;
 
-    EndPoseProblemInitializer init_;
+    EndPoseProblemInitializer parameters;
```
Why no `_`?

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,131 @@
+    void SetJointVelocityLimit(const double& qdot_max_in)
+    {
+        q_dot_max_ = qdot_max_in;
+        xxdiff_max_ = q_dot_max_ * tau_;
```
Change from `xdiff_max` to ``xxdiff_max`` should be reverted

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,131 @@
-    TaskSpaceVector CostPhi;
-    TaskSpaceVector InequalityPhi;
-    TaskSpaceVector EqualityPhi;
+    TaskSpaceVector cost_phi;
```
Upper case Phi

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,131 @@
-    std::vector<TaskSpaceVector> Phi;
-    std::vector<Eigen::MatrixXd> J;
-    std::vector<Hessian> H;
+    std::vector<TaskSpaceVector> phi;
```
Upper case Phi

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,131 @@
-    double qdot_max;   //!< Joint velocity limit (rad/s)
-    double xdiff_max;  //!< Maximum change in the variables in a single timestep tau. Gets set/updated via setTau().
+    double q_dot_max_;   //!< Joint velocity limit (rad/s)
+    double xxdiff_max_;  //!< Maximum change in the variables in a single timestep tau_. Gets set/updated via SetTau().
```
One x too many

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,131 @@
+    double xxdiff_max_;  //!< Maximum change in the variables in a single timestep tau_. Gets set/updated via SetTau().
 
-    double W_rate;  //!< Kinematic system transition error covariance multiplier (constant throughout the trajectory)
+    double w_rate_;  //!< Kinematic system transition error covariance multiplier (constant throughout the trajectory)
```
w != W

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_sampling_problem.h:
```diff
@@ -49,45 +45,42 @@ class TimeIndexedSamplingProblem : public PlanningProblem, public Instantiable<T
-    void setGoalState(Eigen::VectorXdRefConst qT);
-    void setGoalTime(const double& t);
+    Eigen::VectorXd vel_limits;
+    TaskSpaceVector phi;
```
Phi

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_sampling_problem.h:
```diff
@@ -49,45 +45,42 @@ class TimeIndexedSamplingProblem : public PlanningProblem, public Instantiable<T
+    TimeIndexedSamplingProblemInitializer parameters;
+    TaskSpaceVector constraint_phi;
 
-    // TODO(wxm): Make private and expose getter/setter where required.
```
Please retain this comment.

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/problems/unconstrained_end_pose_problem.h:
```diff
@@ -52,34 +45,33 @@ class UnconstrainedEndPoseProblem : public PlanningProblem, public Instantiable<
-    Eigen::MatrixXd J;
-    Hessian H;
-    Eigen::VectorXd qNominal;
+    TaskSpaceVector phi;
```
Phi

wxmerkt commented on exotica_core/include/exotica_core/problems/unconstrained_time_indexed_problem.h:
```diff
@@ -1,109 +1,103 @@
-    std::vector<TaskSpaceVector> Phi;
-    std::vector<Eigen::MatrixXd> J;
-    std::vector<Hessian> H;
+    std::vector<TaskSpaceVector> phi;
```
Phi

wxmerkt commented on exotica_core/include/exotica_core/problems/unconstrained_time_indexed_problem.h:
```diff
@@ -1,109 +1,103 @@
+    int num_tasks;
 
-    TaskSpaceVector CostPhi;  // passed to the TimeIndexedTask, needs to be kept for reinitialisation
+    TaskSpaceVector cost_phi;  // passed to the TimeIndexedTask, needs to be kept for reinitialisation
```
Phi

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/scene.h:
```diff
@@ -221,21 +213,13 @@ class Scene : public Object, Uncopyable, public Instantiable<SceneInitializer>
-    void updateMoveItPlanningScene();
-
-    void loadSceneFromStringStream(std::istream& in, const Eigen::Isometry3d& offset, bool updateCollisionScene);
+    KinematicsRequest kinematic_equest_;
```
request, not equest

wxmerkt replied:
Unresolved

wxmerkt commented on exotica_core/include/exotica_core/server.h:
```diff
@@ -1,37 +1,6 @@
-
-#ifndef EXOTICA_INCLUDE_EXOTICA_SERVER_H_
-#define EXOTICA_INCLUDE_EXOTICA_SERVER_H_
+
```
Copyright message missing

wxmerkt commented on exotica_core/include/exotica_core/tasks.h:
```diff
@@ -49,85 +45,81 @@ namespace exotica
-    int JN;
-    int NumTasks;
-    double Tolerance = 0.0;  // To avoid numerical issues consider all values below this as 0.0.
+    int length_phi;
```
lower case phi here as it's a task map, so correct

wxmerkt commented on exotica_core/include/exotica_core/tasks.h:
```diff
@@ -49,85 +45,81 @@ namespace exotica
+    void Update(const TaskSpaceVector& big_phi, int t);
+    void ReinitializeVariables(int _T, std::shared_ptr<PlanningProblem> _prob, const TaskSpaceVector& _phi);
+
+    std::vector<Eigen::VectorXd> rho;
```
We need to make a decision whether upper or lower case Rho. I think your lower case was a fix of a previously wrong one. But need to review

VladimirIvan replied:
It's lower case in the equations. We'd still have Rho in the XML though.

wxmerkt commented on exotica_core/include/exotica_core/visualization.h:
```diff
@@ -1,37 +1,33 @@
- * POSSIBILITY OF SUCH DAMAGE.
- *
- */
+// Copyright (c) 2018, University of Edinburgh
```
Copyright change

wxmerkt commented on exotica_core/src/kinematic_tree.cpp:
```diff
@@ -46,607 +42,607 @@ namespace exotica
-    KDL::Frame transformKDL;
-    tf::transformEigenToKDL(transform, transformKDL);
-    std::shared_ptr<KinematicElement> NewElement = std::make_shared<KinematicElement>(Tree.size(), parent_element, KDL::Segment(name, KDL::Joint(KDL::Joint::None), transformKDL, inertia));
+    KDL::Frame transform_kld;
```
kld != kdl

wxmerkt commented on exotica_core/src/problems/time_indexed_problem.cpp:
```diff
@@ -38,628 +34,628 @@ REGISTER_PROBLEM_TYPE("TimeIndexedProblem", exotica::TimeIndexedProblem)
-    qdot_max = init_.JointVelocityLimit;
-    applyStartState(false);
-    reinitializeVariables();
+    T_ = init_.t;
```
Ouch. Here our entire automatic camel casing etc. is breaking: We can't automatically do it for mathematical/physical quantities as now the XML/Python would have one meaning and the C++ another. Further, we would require checks that an automatic changed variable isn't shadowing or clashing (I did not review the initializer generation Python in full) - i.e. what if my initialiser has `delta` and `Delta` (because in the paper two different meanings are assigned to them).

wxmerkt commented on exotica_core/src/task_space_vector.cpp:
```diff
@@ -1,40 +1,36 @@
 namespace exotica
 {
-TaskVectorEntry::TaskVectorEntry(int inId_in, RotationType type_in) : inId(inId_in), type(type_in)
+TaskVectorEntry::TaskVectorEntry(int _in_id, RotationType _type) : in_id(_in_id), type(_type)
```
We are inconsistent with how to name variables that are passed in:

- Sometimes it's `_id`
- Other times `id_in`
- And here it even is `_in_id`

I'd opt for `id_in` but either way we need to use one and stick to id

wxmerkt commented on exotica_core/src/tools.cpp:
```diff
@@ -1,63 +1,59 @@
 #include <random>
 #include <regex>
-#include <typeinfo>  // NB: The RTTI Functionality of C++
+#include <typeinfo>  // The RTTI Functionality of C++
```
Few people immediately know what RTTI is

VladimirIvan replied:
`NB:` looks like a doxygen tag and I had had to look it up before realising it's not doxygen. That was way more confusing than RTTI.

wxmerkt commented on exotica_core/src/visualization.cpp:
```diff
@@ -1,40 +1,36 @@
- * POSSIBILITY OF SUCH DAMAGE.
- *
- */
+// Copyright (c) 2018, University of Edinburgh
```
Copyright change

wxmerkt commented on exotica_python/src/pyexotica.cpp:
```diff
@@ -304,27 +333,29 @@ struct type_caster<Initializer>
 
     bool PyToInit(PyObject* source, Initializer& ret)
     {
+        HIGHLIGHT("Check tuple")
```
Do we want those highlights to remain here? (A few more below)

wxmerkt commented on exotica_python/src/pyexotica.cpp:
```diff
@@ -600,283 +632,283 @@ PYBIND11_MODULE(_pyexotica, module)
-        .def_readonly("Phi", &TimeIndexedTask::Phi)
-        // .def_readonly("H", &TimeIndexedTask::H)
-        .def_readonly("J", &TimeIndexedTask::J)
+        .def_readonly("phi", &TimeIndexedTask::phi)
```
Upper case Phi - I mentioned above but also needs to be fixed in Python

VladimirIvan commented:
- Fixed licenses
- changed `i++` to `++i` where it doesn't change the logic.
- Renamed `phi` to `Phi`.
- Renamed `Tau` to `tau`.
- Fixed header guards and `#endif`.
- Renamed `w_rate_` to `w_scale_`.
- `q_parametersin` to `q_parameters_in`.
- `(double)` to `static_cast<double>()`.
- In `TaskVectorEntry`: `in_id` to `id`, `_in_id`, to `_id`, iterator of entries now named `entry` (instead of `id` to avoid confusion).
- Removed debugging `HIGHLIGHT` from python wrapper.
- Removed initializer member case changing - reverted lower case variable naming in c++ Initializer class.

Documentation will be a separate PR.

>> How does (or does it at all) this work for meta initialisers? Is class required?

The `class` tag is still required but will be only used for naming the initializer class, not for instanciating anything.

>> Getters and setters may (or should?) be set_variable instead of SetVariable

While I agree, we have very few pure accessors. The list is below. Each one of them is in a class with anywhere between 1 to 10 methods with Set/Get in the name but is exposing other get/set related functionality (e.g., not exposing a private_member_variable_). Renaming the accessors will make the classes a capital mess. Input welcome.

```
EffAxisAlignmnet:
Set/GetDirection
Set/GetAxis

IMesh
SetWrights

PRM:
Set/GetMultiQuery
Set/GetRange

CollisionScene:
SetACM
Set/GetAlwaysExternallyUpdatedCollisionScene
Set/GetRobotLinkScale
Set/GetWorldLinkScale
Set/GetRobotLinkPadding
Set/GetWorldLinkPadding
Set/GetReplacePrimitiveShapesWithMeshes

MotionSolver:
Set/GetNumberOfMaxIterations
GetPlanningTime
GetProblem

PlanningProblem:
Set/GetStartTime
SetGetTau

Initializer:
Set/GetName
```

wxmerkt commented on exotations/exotica_core_task_maps/src/interaction_mesh.cpp:
```diff
@@ -39,42 +35,42 @@ namespace exotica
     Eigen::MatrixXd dist;
     Eigen::VectorXd wsum;
-    Eigen::VectorXd eff_Phi(M * 3);
+    Eigen::VectorXd eff_phi(M * 3);
```
Unresolved

wxmerkt commented on exotations/exotica_core_task_maps/src/joint_velocity_limit.cpp:
```diff
@@ -1,44 +1,40 @@
- * POSSIBILITY OF SUCH DAMAGE.
- *
- */
+// Copyright (c) 2018, University of Edinburgh
```
Unresolved

wxmerkt commented on exotations/exotica_core_task_maps/src/point_to_plane.cpp:
```diff
@@ -1,111 +1,107 @@
-#include "exotica_core_task_maps/point_to_plane.h"
-
-REGISTER_TASKMAP_TYPE("Point2Plane", exotica::Point2Plane);
+// Copyright (c) 2018, University of Edinburgh
```
Unresolved

wxmerkt commented on exotations/exotica_core_task_maps/src/point_to_plane.cpp:
```diff
@@ -117,5 +113,4 @@ void Point2Plane::publish_debug()
 }
-
-}  // namespace exotica
+}
```
Unresolved

wxmerkt commented on exotations/solvers/exotica_aico_solver/src/aico_solver.cpp:
```diff
@@ -58,21 +54,21 @@ void AICOSolver::Instantiate(AICOSolverInitializer& init)
-    minimum_step_tolerance_ = init.MinStep;
-    step_tolerance_ = init.StepTolerance;
-    function_tolerance_ = init.FunctionTolerance;
+    minimum_step_tolerance = init.MinStep;
```
Need to be reverted, see comments in header (still unresolved)

wxmerkt commented on exotations/solvers/exotica_levenberg_marquardt_solver/include/exotica_levenberg_marquardt_solver/levenberg_marquardt_solver.h:
```diff
@@ -1,9 +1,39 @@
+//
```
I think the LM solver is MIT licensed, i.e., this is incorrect:
https://github.com/ipab-slmc/exotica/blob/bb96e3a8338ef64fe7243c215457ac036f9d57ec/exotations/solvers/exotica_levenberg_marquardt_solver/package.xml#L9

wxmerkt commented on exotations/solvers/exotica_levenberg_marquardt_solver/src/levenberg_marquardt_solver.cpp:
```diff
@@ -1,18 +1,47 @@
+//
```
Incorrect, this solver is MIT licensed

wxmerkt commented on exotica_core/cmake/generate_initializers.py:
```diff
@@ -2,228 +2,276 @@
 import os
+import re
+
+def to_camel_cased(name):
```
To be removed as not used anymore

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +75,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
-    ArrayTwist PhiDot;
-    ArrayJacobian J;
-    ArrayJacobian JDot;
+    ArrayTwist phi_dot;
```
Unresolved. If Phi is upper case, Phi_dot should be too.

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -78,209 +75,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
-    Eigen::Map<ArrayTwist> PhiDot{nullptr, 0};
-    Eigen::Map<ArrayJacobian> J{nullptr, 0};
-    Eigen::Map<ArrayJacobian> JDot{nullptr, 0};
+    Eigen::Map<ArrayTwist> phi_dot{nullptr, 0};
```
As above, needs to be upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_end_pose_problem.h:
```diff
@@ -52,31 +47,31 @@ class BoundedEndPoseProblem : public PlanningProblem, public Instantiable<Bounde
-    int PhiN;
-    int JN;
-    int NumTasks;
+    int length_phi;
```
Upper case... - applies too all problems!

wxmerkt commented on exotica_core/include/exotica_core/problems/bounded_time_indexed_problem.h:
```diff
@@ -1,105 +1,100 @@
-    int PhiN;
-    int JN;
-    int NumTasks;
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/sampling_problem.h:
```diff
@@ -48,46 +46,42 @@ class SamplingProblem : public PlanningProblem, public Instantiable<SamplingProb
+    SamplingTask inequality;
+    SamplingTask equality;
+
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,132 @@
-    TaskSpaceVector InequalityPhi;
-    TaskSpaceVector EqualityPhi;
+    TaskSpaceVector cost_phi;
+    TaskSpaceVector inequality_phi;
```
As above

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,132 @@
-    TaskSpaceVector EqualityPhi;
+    TaskSpaceVector cost_phi;
+    TaskSpaceVector inequality_phi;
+    TaskSpaceVector equality_phi;
```
As above

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_problem.h:
```diff
@@ -1,137 +1,132 @@
-    int JN;
-    int NumTasks;
-    bool useBounds;
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/time_indexed_sampling_problem.h:
```diff
@@ -49,49 +46,46 @@ class TimeIndexedSamplingProblem : public PlanningProblem, public Instantiable<T
-    int PhiN;
-    int JN;
-    int NumTasks;
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/unconstrained_end_pose_problem.h:
```diff
@@ -52,34 +46,33 @@ class UnconstrainedEndPoseProblem : public PlanningProblem, public Instantiable<
-    int NumTasks;
-
-    int getTaskId(const std::string& task_name);
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/include/exotica_core/problems/unconstrained_time_indexed_problem.h:
```diff
@@ -1,109 +1,104 @@
-    int JN;
-    int NumTasks;
+    bool IsValid() override { return true; }
+    int length_phi;
```
Upper case

wxmerkt commented on exotica_core/src/problems/bounded_end_pose_problem.cpp:
```diff
@@ -1,67 +1,65 @@
-    JN = 0;
-    for (int i = 0; i < NumTasks; i++)
+    num_tasks = tasks_.size();
+    length_phi = 0;
```
Upper case

wxmerkt commented on exotica_python/src/pyexotica.cpp:
```diff
@@ -600,283 +629,283 @@ PYBIND11_MODULE(_pyexotica, module)
-        .def_readonly("PhiN", &TimeIndexedTask::PhiN)
-        .def_readonly("JN", &TimeIndexedTask::JN)
-        .def_readonly("num_tasks", &TimeIndexedTask::NumTasks)
+        .def_readonly("length_phi", &TimeIndexedTask::length_phi)
```
Upper case Phi, I think?

wxmerkt commented:
I've resolved most of the issues apart from:

- Phi vs phi naming (see comment above:
  (a) **Phi**: Forward kinematic mapping
  (b) phi: value of individual taskmap
  (c) Phi: value of all taskmaps, i.e., Phi=(phi_0, phi_1, phi_2)
- The change (possible bug) introduced into `TimeIndexedRRTConnect` by changing T to goal time
- `generate_initialisers.py` changes: Do some of them have to be reverted?
- Some minor other changes

wxmerkt commented:
Comments to go through (I have resolved all others by new commits):

- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244647062
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244647121
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244647163
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244647177
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244951225
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244955487
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244955595
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244955767
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244956764
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957022
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957500
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957547
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957592
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957648
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957823
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957877
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244957963
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244958103
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244958138
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244959851
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244961263
- [ ] https://github.com/ipab-slmc/exotica/pull/475#discussion_r244646883

VladimirIvan commented:
- I have fixed `*_phi` members.
- Added `catkin_EXPORTED_TARGETS` dependencies into cmake.
- Moved headers not used in the class definition into the cpp. Check depending code. This revealed assumed include dependencies in task maps and solvers. It will be probably much worse in other repos.

wxmerkt commented on exotica_core/include/exotica_core/kinematic_tree.h:
```diff
@@ -32,21 +32,17 @@
 #include <kdl/jacobian.hpp>
 #include <kdl/tree.hpp>
-#include <kdl_parser/kdl_parser.hpp>
+#include <tf/transform_datatypes.h>
```
We should move to tf2 soon
